added rspack support (#254)
Trackunit/developer-hub: changelogs/upgrade-to-rspack-from-webpack.md

---
title: trackunit/iris-app@v0.0.NEXT - upgrade to rspack from webpack
type: improved
---

In this version of our Iris App SDK we have added support for Rspack as well as the existing Webpack.

> > to read more on rspack, please visit the [rspack documentation](https://rspack.dev/).

Author: yes a call out box

> Iris apps created with `@trackunit/iris-app@0.0.NEXT` should already have the required rspack configuration.

### Rename and update the webpack config file

Rename the `webpack.config.ts` file to `rspack.config.ts` and update the file to use the rspack configuration.

> for the new `rspack.config.ts` file, you shold change the following import.

Reviewer: ```suggestion
For the new `rspack.config.ts` file, you shold change the following import.
```

```
import { Configuration } from "webpack";
```
Change to: 
```
import { Configuration } from "@rspack/core";
```

> The end result should look like the following.

Reviewer: ```suggestion
The end result should look like the following:
```

```
import { Configuration } from "@rspack/core";

export default (configuration: Configuration) => {
  return configuration;
};
```

> ### Change the following in `targets` list of the `project.json` file of your iris app.

Reviewer: ```suggestion
### Change the following entries in `targets` map in the `project.json` file of your iris app.
```

> for `build` target, change the following:

Reviewer: ```suggestion
For the `build` target, change the following:
```

```
    "executor": "@trackunit/iris-app:build",
```
Change to: 
```
    "executor": "@trackunit/iris-app-sdk-rspack:build",
```

> for `serve` target, change the following:

Reviewer: ```suggestion
For the `serve` target, change the following:
```
